pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1314,13 +1314,10 @@ private Topic getTopicReference(TopicName topicName) {
                         TopicName partitionTopicName = TopicName.get(topicName.getPartitionedTopicName());
                         PartitionedTopicMetadata partitionedTopicMetadata = getPartitionedTopicMetadata(partitionTopicName, false);
                         if (partitionedTopicMetadata == null || partitionedTopicMetadata.partitions == 0) {
-                        	final String errSrc;
-                        	if (partitionedTopicMetadata != null) {
-                        		errSrc = " has zero partitions";
-                        	} else {
-                        		errSrc = " has no metadata";
-                        	}
-                            return new RestException(Status.NOT_FOUND, "Partitioned Topic not found: " + topicName.toString() + errSrc);
+                            final String errSrc = partitionedTopicMetadata != null ?
+                                    "has zero partitions" : "has no metadata";
+                            return new RestException(Status.NOT_FOUND, String.format(
+                                    "Partitioned Topic not found: %s - %s", topicName.toString(), errSrc));
```

Comment:
nit: the error message here can also be improved. if `partitionedTopicMetadata == null`, then it is `Topic not found`, otherwise it is `Partitioned Topic not found`.

Reply:
I've just refactored this code by a simplified version, please take a look now :+1: